Add customizable properties

rikusouda opened the pull request:
* Add following properties
    * ~~If `isButtonHeightFlexible` is enabled,~~ button's height will same with `CBreadcrumbControl`
    * You can set `buttonFont` to change font of item buttons
    * You can create `CBreadcrumbControl` from code without xib or Storyboard using `init(frame:)`
* Clean up some codes

For the time being, other ideas will set as issue.

apparition47 commented on BreadCrumb Control/BreadCrumb Control/BreadCrumb.swift:
```diff
@@ -170,6 +182,14 @@ public class CBreadcrumbControl: UIScrollView {
     }
 
     
+    private var buttonHeight: CGFloat {
```
`frame.size.height` is great. I've been thinking about the `kBreadcrumbHeight` constant and I wonder if there's a usecase for someone to have a fixed height even though the `frame.size.height` is larger. Perhaps `kBreadcrumbHeight` isn't needed? What do you think?

rikusouda replied:
That's good discussion.
I think, if someone needs fixed height, maybe it needs other customize. So we don't have to provide fixed height. 

So, I'll delete `kBreadcrumbHeight` and `isButtonHeightFlexible`. How about?

apparition47 replied:
I think that's the best option!

apparition47 replied:
Does it still need to be a computed property?

rikusouda replied:
Oops, it's no longer needed.

apparition47 commented:
Unfortunately, as of XCode 9 beta, `@IBInspectable` properties don't work for `enum` and `UIFont` and only work for:

* `Int`
* `CGFloat`
* `Double`
* `String`
* `Bool`
* `CGPoint`
* `CGSize`
* `CGRect`
* `UIColor`
* `UIImage`

I've seen some people do [some workarounds](https://stackoverflow.com/questions/43089274/swift-uifont-ibinspectable-is-it-possible) using `Int` or `String` mappings to a `@IBInspectable` property, but I'm not a big fan of their approach. 

Probably best to remove the `@IBInspectable` decorator for those two properties and allow the user to access the property in code. Documentation would be helpful here.

Thanks, @rikusouda

rikusouda commented:
@apparition47 
Thanks.  I didn't know that, because I do not use InterfaceBuilder much.
I apply your idea to this PR. And I fixed `Readme.md` and cleaned up codes to make readable it.

rikusouda commented:
I think the index (+1) at touch up is not good. In the case of the Root button, I think that handlers should be separated. What do you think, @apparition47.
(I'll create new PR about this Issue, if you want.)

apparition47 commented:
I definitely agree. I think it deserves a separate PR.

I'll review this PR again when I get home!

And thanks again for your work on this!

rikusouda commented:
Thanks your great review at every day (night?).
I'll create new PR after this PR will be merged.

apparition47 commented:
I live in Tokyo too, so it's always nighttime!

And your changes look very good. Thanks!

rikusouda commented:
Thanks!

> I live in Tokyo too

Sounds cool 👍
